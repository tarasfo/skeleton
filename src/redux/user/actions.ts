import {action} from '../helpers';
import * as types from './constants';

export const requestUser = () =>  action(types.REQUEST_USER);

export const setUser = (user: any) =>  action(types.SET_USER, user);