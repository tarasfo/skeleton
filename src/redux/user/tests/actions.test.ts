import {requestUser, setUser} from '../actions';

const user = {
   field1: 'value1',
};

describe('request user action test', () => {
   it('dispatches request user action', () => {
      expect(requestUser()).toMatchSnapshot();
   });
});
describe('set user action test', () => {
    it('dispatches set user action', () => {
       expect(setUser(user)).toMatchSnapshot();
    });
 });