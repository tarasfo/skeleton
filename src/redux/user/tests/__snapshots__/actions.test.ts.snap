// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request user action test dispatches request user action 1`] = `
Object {
  "payload": undefined,
  "type": "user/REQUEST_USER",
}
`;

exports[`set user action test dispatches set user action 1`] = `
Object {
  "payload": Object {
    "field1": "value1",
  },
  "type": "user/SET_USER",
}
`;
